packages/sanity/src/core/store/bundles/createBundlesMetadataAggregator.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  bufferTime,
+  catchError,
+  concat,
+  EMPTY,
+  filter,
+  iif,
+  merge,
+  type Observable,
+  of,
+  switchMap,
+} from 'rxjs'
+import {type SanityClient} from 'sanity'
+
+import {type BundlesMetadata} from '../../releases/tool/useBundlesMetadata'
+
+export type BundlesMetadataMap = Record<string, BundlesMetadata>
+
+export type MetadataWrapper = {data: BundlesMetadataMap | null; error: null; loading: boolean}
+
+const getFetchQuery = (bundleSlugs: string[]) => {
+  // projection key must be string - cover the case that a bundle has a number as first char
+  const getSafeSlug = (slug: string) => `bundle_${slug.replaceAll('-', '_')}`
+
+  return bundleSlugs.reduce(
+    ({subquery: accSubquery, projection: accProjection}, bundleSlug) => {
+      const safeSlug = getSafeSlug(bundleSlug)
+
+      const subquery = `${accSubquery}"${safeSlug}": *[_id in path("${bundleSlug}.*")]{_updatedAt } | order(_updatedAt desc),`
+
+      // conforms to BundlesMetadata
+      const projection = `${accProjection}"${bundleSlug}": {
+              "updatedAt": ${safeSlug}[0]._updatedAt,
+              "documentCount": count(${safeSlug})
+            },`
+
+      return {subquery, projection}
+    },
+    {subquery: '', projection: ''},
+  )
+}
+
+/**
+ * @internal
+ *
+ * An initial fetch is made. This fetch is polled whenever a listener even is emitted
+ * Only bundles that have been mutated are re-fetched
+ *
+ * @returns an Observable that accepts a list of bundle slugs and returns a stream of metadata
+ */
+export const createBundlesMetadataAggregator = (client: SanityClient | null) => {
+  const aggregatorFetch$ = (
+    bundleSlugs: string[],
+    isInitialLoad: boolean = false,
+  ): Observable<MetadataWrapper> => {
+    if (!bundleSlugs?.length || !client) return EMPTY
+
+    const {subquery: queryAllDocumentsInBundleSlugs, projection: projectionToBundleMetadata} =
+      getFetchQuery(bundleSlugs)
+
+    const initialLoading$ = of({loading: true, data: null, error: null})
+    const fetchData$ = client.observable
+      .fetch<BundlesMetadataMap>(
+        `{${queryAllDocumentsInBundleSlugs}}{${projectionToBundleMetadata}}`,
+      )
+      .pipe(
+        switchMap((response) => of({data: response, error: null, loading: false})),
+        catchError((error) => {
+          console.error('Failed to fetch bundle metadata', error)
+          return of({data: null, error, loading: false})
+        }),
+      )
+
+    // initially emit loading empty state if first fetch
+    return iif(() => isInitialLoad, concat(initialLoading$, fetchData$), fetchData$)
```

Comment:
Have you taken a look at [the `startsWith` operator](https://rxjs.dev/api/operators/startWith)? If it fits your use case, it _might_ be a simpler way to compose the initial value into the observable. We use it in a few places already.